R/accessory.R

```diff
@@ -956,10 +956,14 @@ read_contrasts <-
       # Extract blocking factors from 'formula' if available
       blocking <- NA
       formula <- NA
+      make_contrasts_str <- NA
       if (!is.null(x$formula)) {
-        terms <- trimws(unlist(strsplit(gsub("~", "", x$formula), "\\+")))
+        formula_obj <- as.formula(x$formula)
+        terms_obj <- terms(formula_obj)
+        terms <- attr(terms_obj, "term.labels")
```

Comment:
Here I changed the way that `terms` are obtained, as the previous method with `(gsub("~", "", x$formula)` did not took into account that formulas could have `~ 0 + term + blocking variable`, therefore, 0 would be added to blocking `blocking_vars`.